packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -428,6 +445,24 @@ describe('<DataGrid /> - Layout & Warnings', () => {
         'The data grid component requires all rows to have a unique id property',
       );
     });
+
+    it('should throw if the parent container has no intrinsic height', function test() {
+      // TODO is this fixed?
+      if (!/jsdom/.test(window.navigator.userAgent)) {
+        // can't catch render errors in the browser for unknown reason
+        // tried try-catch + error boundary + window onError preventDefault
+        this.skip();
+      }
+
+      expect(() => {
+        render(
+          <div>
+            <DataGrid {...baselineProps} />
+          </div>,
+        );
+        // @ts-expect-error need to migrate helpers to TypeScript
+      }).toWarnDev();
+    });
```

Comment:
That was the test from the previous PR. I thought changing it would make sense because there is also a check that if the parent container doesn't have height a warning should be thrown and since that wasn't happening when the height increases indefinitely I thought checking for that makes sense.

Reply:
Ok, how about we add this proposed test? I think that we still need to hedge against #344 in the future. I don't want to be the last line of defense against this regression 😁.

Reply:
I'm also in favour of adding a specific test to cover this. I'll update it.